src/subtype.c

```diff
@@ -2444,11 +2444,22 @@ static jl_value_t *intersect_all(jl_value_t *x, jl_value_t *y, jl_stenv_t *e)
     e->Runions.more = 0;
     memset(e->Runions.stack, 0, sizeof(e->Runions.stack));
     jl_value_t **is;
-    JL_GC_PUSHARGS(is, 2);
+    JL_GC_PUSHARGS(is, 3);
+    jl_value_t **saved = &is[2];
+    jl_savedenv_t se;
+    save_env(e, saved, &se);
     int lastset = 0, niter = 0, total_iter = 0;
     jl_value_t *ii = intersect(x, y, e, 0);
+    if (ii == jl_bottom_type) {
+        restore_env(e, *saved, &se);
+    }
+    else {
+        free(se.buf);
+        save_env(e, saved, &se);
+    }
     while (e->Runions.more) {
         if (e->emptiness_only && ii != jl_bottom_type) {
+            free(se.buf);
```

Comment:
Somebody should add a language facility to C to call a certain function automatically at the end of a scope.

Reply:
Maybe give it some kind of obscure acronym?